src/messages.cpp

```diff
@@ -473,11 +473,18 @@ Messages::dialog::dialog()
 
 void Messages::dialog::init( ui_adaptor &ui )
 {
-    w_width = std::max( 45, TERMX - 2 * ( panel_manager::get_manager().get_width_right() +
-                                          panel_manager::get_manager().get_width_left() ) );
+    const int left_panel_width = panel_manager::get_manager().get_width_left();
+    const int right_panel_width = panel_manager::get_manager().get_width_right();
     w_height = TERMY;
-    w_x = ( TERMX - w_width ) / 2;
-    w_y = ( TERMY - w_height ) / 2;
+    w_y = 0;
+    // try to center and not obscure sidebar
+    w_x = std::max( left_panel_width, right_panel_width );
+    w_width = TERMX - 2 * w_x;
+    if( w_width < w_height * 3 ) {
```

Comment:
Did you mean 
```c++
if( w_width * 3 < w_height ) {
```
here?

Reply:
It tests if the width-to-height ratio is lower than 1.5 (meaning the window is too narrow, assuming that the font width-to-height ratio is 0.5) and increase the width, so it's not wrong.

Reply:
If I remember correctly window width and height are number of characters, irrespective of font size.

Reply:
That's right, but as I said in the OP

> It currently does not take font size into account in the SDL version, because font size is unknown in the curses version.

Reply:
Thank you for clarification. I didn't expect that you set width-to-height threshold to 1.5. I thought you were assuming width-to-height ratio below 0.5 as "too narrow".